homeassistant/components/netatmo/climate.py

```diff
@@ -349,6 +349,7 @@ def __init__(self, auth, home=None):
         self.schedules = []
         self.home = home
         self.home_id = None
+        self.setup()
```

Comment:
We try to avoid side effects in init method.

Reply:
Not sure if this is really better as the id needs to get initialised eventually.

Reply:
Why not just move the call of `setup` to just after creating the instance?

https://github.com/home-assistant/home-assistant/blob/9e6732e53002a54489fff86efc5cd00c881e220c/homeassistant/components/netatmo/climate.py#L113

Reply:
Fine with me.